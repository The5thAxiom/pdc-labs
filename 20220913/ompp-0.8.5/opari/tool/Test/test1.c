#include <stdio.h>
#ifdef _OPENMP
   #include <omp.h>  //just testing
#endif

int main() {
  omp_lock_t l;

  printf("before...\n");

  #pragma omp parallel
  {{
    printf("parallel 1...\n");
  }} //end

  //#pragma omp parallel
  {
    //printf("parallel 1...\n");
  }

/*
  #pragma omp parallel
  {
    printf("parallel 1...\n");
  }
*/

  # /*complicated*/ pragma \
                    omp \
    /*more*/        parallel
  {
    printf("parallel 2...\n");
  }

  //omp_init_lock();
  omp_init_lock(&l);

  printf("after...\n");
}
