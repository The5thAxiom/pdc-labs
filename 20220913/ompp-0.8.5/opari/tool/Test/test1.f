      program test1
      gg = 9
c     comment
c$OMP parallel
*$oMp barrier
!$OmP barrier
c $omp no
!!$omp mo
!$OmP end
!$OmP+
!$OmP+parallel
!$OmP atomic
      gg = 7
!$OmP    parallel
!$OmP&   default(shared)
      gg = 7
!$OmP    end
!$OmP+   parallel
      end
