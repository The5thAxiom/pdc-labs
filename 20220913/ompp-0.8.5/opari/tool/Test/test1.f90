      program test1
      gg = 9
       !$Omp parallel
       !  !$Omp no
      gg = 7
       !$OMP end          &
       !$OMP parallel
       !$OMP parallel     &
       !$OMP &  default   &
       !$OMP &  (shared)
       !$OMP end          &     !extra comment
       !$OMP parallel
      end
