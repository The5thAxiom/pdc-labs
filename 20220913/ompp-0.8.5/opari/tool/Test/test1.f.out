#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#line 1 "test1.f"
      program test1
      gg = 9
c     comment
      call POMP_Parallel_fork(46)
#line 4 "test1.f"
c$omp parallel
      call POMP_Parallel_begin(46)
#line 5 "test1.f"
      call POMP_Barrier_enter(47)
#line 5 "test1.f"
*$omp barrier
      call POMP_Barrier_exit(47)
#line 6 "test1.f"
      call POMP_Barrier_enter(48)
#line 6 "test1.f"
!$omp barrier
      call POMP_Barrier_exit(48)
#line 7 "test1.f"
c $omp no
!!$omp mo
      call POMP_Barrier_enter(46)
!$omp barrier
      call POMP_Barrier_exit(46)
      call POMP_Parallel_end(46)
#line 9 "test1.f"
!$omp end
!$omp+
!$omp+parallel
      call POMP_Parallel_join(46)
#line 12 "test1.f"
      call POMP_Atomic_enter(49)
#line 12 "test1.f"
!$omp atomic
      gg = 7
      call POMP_Atomic_exit(49)
#line 14 "test1.f"
      call POMP_Parallel_fork(50)
#line 14 "test1.f"
!$omp    parallel
!$omp&   default(shared)
      call POMP_Parallel_begin(50)
#line 16 "test1.f"
      gg = 7
      call POMP_Barrier_enter(50)
!$omp barrier
      call POMP_Barrier_exit(50)
      call POMP_Parallel_end(50)
#line 17 "test1.f"
!$omp    end
!$omp+   parallel
      call POMP_Parallel_join(50)
#line 19 "test1.f"
      end
